concrete/src/Job/JobFactory.php

```diff
@@ -0,0 +1,278 @@
+<?php
+namespace Concrete\Core\Job;
+
+use Concrete\Core\Application\Application;
+use Concrete\Core\Package\PackageList;
+
+class JobFactory
+{
+    /** @var \Concrete\Core\Application\Application */
+    private $app;
+
+    public function __construct(Application $app)
+    {
+        $this->app = $app;
+    }
+
+    /**
+     * Return list of installed Jobs.
+     *
+     * @return Job[]
+     */
+    public function installed()
+    {
+        $jobs = [];
+
+        $db = $this->app['database']->connection();
+        $q = "SELECT jID FROM Jobs ORDER BY jDateLastRun, jID";
+        $r = $db->query($q);
+
+        while ($row = $r->fetchRow($q)) {
+            $j = $this->getByID($row['jID']);
+            if (is_object($j)) {
+                $jobs[] = $j;
+            }
+        }
+
+        return $jobs;
+    }
+
+    /**
+     * Return list of scheduled Jobs.
+     *
+     * @return Job[]
+     */
+    public function scheduled()
+    {
+        $jobs = [];
+
+        $db = $this->app['database']->connection();
+        $q = "SELECT jID FROM Jobs WHERE isScheduled = 1 ORDER BY jDateLastRun, jID";
+        $r = $db->query($q);
+
+        while ($row = $r->fetchRow($q)) {
+            $j = $this->getByID($row['jID']);
+            if (is_object($j)) {
+                $jobs[] = $j;
+            }
+        }
+
+        return $jobs;
+    }
+
+    /**
+     * @param int $jID
+     *
+     * @return null|Job
+     */
+    public function getByID($jID)
+    {
+        $db = $this->app['database']->connection();
+        $jobData = $db->fetchAssoc("SELECT * FROM Jobs WHERE jID=?", [
+            intval($jID),
+        ]);
+
+        if (!$jobData || !$jobData['jHandle']) {
+            return null;
+        }
+
+        return $this->getJobObjByHandle($jobData['jHandle'], $jobData);
+    }
+
+    /**
+     * @param string $jHandle
+     *
+     * @return null|Job
+     */
+    public function getByHandle($jHandle)
+    {
+        $db = $this->app['database']->connection();
+        $jobData = $db->fetchAssoc("SELECT * FROM Jobs WHERE jHandle=?", [
+            $jHandle,
+        ]);
+
+        if (!$jobData || !$jobData['jHandle']) {
+            return null;
+        }
+
+        return $this->getJobObjByHandle($jobData['jHandle'], $jobData);
+    }
+
+    /**
+     * @param string $jHandle
+     * @param array  $jobData
+     *
+     * @return null|Job
+     */
+    public function getJobObjByHandle($jHandle = '', $jobData = [])
```

Comment:
I still don't think this method should be here, I see that it's used an an internal method to populate the job object, but it that's the case it should be protected or private.